src/askui/agent.py

```diff
@@ -294,30 +303,38 @@ def mouse_scroll(
         self._reporter.add_message("User", f'mouse_scroll: "{x}", "{y}"')
         self.tools.os.mouse_scroll(x, y)
 
-    @telemetry.record_call(exclude={"text"})
-    @validate_call
+    @telemetry.record_call(exclude={"text", "locator"})
+    @validate_call(config=ConfigDict(arbitrary_types_allowed=True))
     def type(
         self,
         text: Annotated[str, Field(min_length=1)],
+        locator: str | Locator | None = None,
+        model: ModelComposition | str | None = None,
     ) -> None:
         """
-        Types the specified text as if it were entered on a keyboard.
+        Types the specified text as if it were entered on a keyboard, optionally triple clicking on the UI element identified to give it focus and select the current text (in multi-line inputs like textareas the current line or paragraph) by the provided locator before typing.
 
         Args:
             text (str): The text to be typed. Must be at least `1` character long.
+            locator (str | Locator | None, optional): The identifier or description of the element (e.g., input field) to type into. If `None`, types at the current focus.
+            model (ModelComposition | str | None, optional): The composition or name of the model(s) to be used for locating the element to type into using the `locator`.
 
         Example:
             ```python
             from askui import VisionAgent
 
             with VisionAgent() as agent:
-                agent.type("Hello, world!")  # Types "Hello, world!"
-                agent.type("user@example.com")  # Types an email address
-                agent.type("password123")  # Types a password
+                agent.type("Hello, world!")  # Types "Hello, world!" at current focus
+                agent.type("user@example.com", locator="Email")  # Clicks on "Email" input, then types
+                agent.type("password123", locator="Password field", model="custom_model")  # Uses specific model
             ```
         """
-        self._reporter.add_message("User", f'type: "{text}"')
-        logger.debug("VisionAgent received instruction to type '%s'", text)
+        msg = f'type "{text}"'
+        if locator is not None:
+            msg += f" into {locator}"
+            self._click(locator=locator, button="left", repeat=3, model=model)
```

Comment:
@programminx-askui I think a better way than triple clicking would be to select the whole content using shortcut but for that I think we need to determine the os beforehand as the shortcut is different for windows/linux (ctl+a) and darwin (command+a). Is this fine for now from your perspective?